src/mappers/country/country_mapper_creator.cpp

```diff
@@ -39,46 +41,137 @@ std::map<std::string, std::string> ImportMappingRules(std::string_view country_m
    return country_mapping_rules;
 }
 
-}  // namespace
-
-
-mappers::CountryMapper mappers::CreateCountryMappings(std::string_view country_mappings_file,
-    const std::map<int, vic3::Country>& countries)
+/// Class so that different mapping strategies can share global variables like country_mappings and
+/// country_mapping_rules
+class CountryMappingCreator
 {
-   std::map<int, std::string> country_mappings;
+   std::map<int, std::string> country_mappings = {};
```

Comment:
member variables should include an underscore at the end of the name to make them easy to identify at a glance. So, ```country_mappings_```